azurerm/internal/services/eventhub/eventhub_consumer_group_data_source.go

```diff
@@ -23,7 +23,7 @@ func dataSourceEventHubConsumerGroup() *schema.Resource {
 			"name": {
 				Type:         schema.TypeString,
 				Required:     true,
-				ValidateFunc: azure.ValidateEventHubConsumerName(),
+				ValidateFunc: azure.ValidateEventHubConsumerDataSourceName(),
```

Comment:
Since we're catering for one specific string case, it might be better to do:
```suggestion
				ValidateFunc: validation.Any(
					azure.ValidateEventHubConsumerName(),
					validation.StringInSlice([]string{"$Default"},
					)
```
Or modify your new validation function to do similar?

Reply:
Ah, didn't know it existed. This is indeed much better. I'll commit your suggestion and remove the rest.